youtube_dl/extractor/thisvid.py

```diff
@@ -0,0 +1,179 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+import itertools
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_str,
+)
+from ..utils import (
+    clean_html,
+    get_element_by_class,
+    sanitize_url,
+    url_or_none,
+    urljoin,
+)
+
+
+class ThisVidIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?thisvid\.com/(?P<type>videos|embed)/(?P<id>[A-Za-z0-9-]+/?)'
```

Comment:
Untested, but this should catch the `//playlist/nnn/video/video-id` pattern.
```suggestion
    _VALID_URL = r'https?://(?:www\.)?thisvid\.com/(?P<type>videos|embed|(?P<pl>playlist))(?(pl)/\d+/video)/(?P<id>[A-Za-z0-9-]+/?)'
```
Full handling of playlists needs a separate set of changes, because the playlist format is a page with a video and a playlist. The yt-dl solution to this is to extract the playlist unless `--no-playlist`, so this pattern has to be caught in the playlist extractor and rewritten to be passed to this extractor in case `--no-playlist`.

Reply:
okay, maybe @afterdelight could test some links

Reply:
here are some examples

https://thisvid.com/playlists/

Reply:
> The playlists are public

i am not aware if there are privates playlist, and how could someone get the link